### PR TITLE
Default impression and success functions added in contribution utilities

### PR DESCRIPTION
**Pull request by Mullefa**
cc @guardian/contributions 

## What does this change?

Currently the `impression()` function for a test created by `contributionUtilities.makeTest()` is inferred from the respective function provided in the options. If this don't exist, but the boolean flag `insertOnView` equals `true`, then an impressions function is created which executes the passed callback once the Epic is inserted. However, there is no default, meaning the test will not have an impression function if neither of these cases occur. This behaviour applies to the `success()` function too, though the semantics are different.

This pull request adds defaults for the `impression()` and `success()` functions that match informally what we have been considering as defaults. However, as raised previously, it is worth discussing formally what these should be at some point, and even whether it makes sense to have configurable `impression()` and `success()` functions for the Epic.

### CODE REVIEW

**Comment by gustavpursche** [comment]
I've just updated `prettier`, which takes care of the code formatting. Could you please rebase/ merge master, before merging this in, to avoid broken checks?

Steps:
- Merge master
- Run `make fix` and commit in case there are changes

Thanks 🙏

**Comment by jranks123** [comment]
👍

**Comment by sndrs** [comment]
any chance we could merge this?

**Comment by Mullefa** [comment]
cc @guardian/contributions - anyone fancy approving this ?

**Review by joelochlann — APPROVED** [review]
This is great - thanks @Mullefa. I always thought that `registerListener` function was more general than we needed, and confusing as a result

Looks like the only test which is overriding these right now is the thrasher one (which is a great example of something which probably shouldn't be using the A/B testing framework at all) - so once we remove this and provide an easy way of using the contribution utilities without being tied to the A/B testing framework, we could probably remove the ability to override these

**Comment by Mullefa** [comment]
@joelochlann yeah - agree with removing the override option in the future. I think we should try and get to a place where the `complete` field is semantically the same over all channels we use (e.g. Epic, Engagement Banner, Thrasher), as this will facilitate querying how a test has performed, as well as combining results of multiple tests and/or channels and/or teams. View this as a (very) small step to that end.

**Comment by joelochlann** [comment]
Actually I'm converting `contributions-utilities.js` to ES6 so maybe we could hold off on merging this

**Comment by Mullefa** [comment]
Sure - if I close this pull request, are you ok to re-implement it in your ES6 conversion work?

**Comment by joelochlann** [comment]
Yeah that sounds like a plan - thanks for this, I will ensure it doesn't get lost!

**Pull request closed without merging**
